kotti/resources.py

```diff
@@ -801,7 +844,13 @@ def default_get_root(request=None):
             this will be an instance of :class:`~kotti.resources.Document`.
     """
 
-    return DBSession.query(Node).filter(Node.parent_id == None).one()  # noqa
+    # Get the root cache, make one if it doesn't exist yet.
+    try:
+        drc = get_settings()['kotti.default_root_cache']
+    except KeyError:
+        drc = get_settings()['kotti.default_root_cache'] = DefaultRootCache()
+
+    return drc.get_root()
```

Comment:
No, the SQLAlchemy object isn't cached.  What **is** cached is the id of the root node.  This allows getting the root node by its primary key which in turn allows SQLAlchemy to cache the root node per session (which should be thread safe TTBOMK).